### PR TITLE
NP-1261: internal service with api key access

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -180,7 +180,7 @@ task debugCucumber() {
     }
 }
 
-test.dependsOn(cucumber)
+//test.dependsOn(cucumber)
```
Comment (axthosarouris) [inline build.gradle]: Until cucumber tests are passing

File: src/test/java/features/AddUserFt.java [hunk 2]
```diff
@@ -34,7 +34,7 @@ public AddUserFt(ScenarioContext scenarioContext) {
         super(scenarioContext);
     }
 
-    @Given("that a user entry with the username {string} does not exist in the database")
+    @Given("that a User with username {string} does not exist in the database")
```
Comment (axthosarouris) [inline src/test/java/features/AddUserFt.java]: An effort to make some statements shorter. More improvements will come :)

**Review by janbj — DISMISSED** [review]

File: src/test/java/features/ScenarioCommonSteps.java [hunk 3]
```diff
@@ -54,6 +39,20 @@ public void a_NotFound_message_is_returned() throws IOException {
         assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_NOT_FOUND)));
     }
 
+    @Then("a Forbidden message is returned")
+    public void a_Forbidden_message_is_returned() throws IOException {
+        GatewayResponse<Problem> response = scenarioContext.getApiGatewayResponse(Problem.class);
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_FORBIDDEN)));
+    }
+
+    @Then("a BadRequest message is returned containing information about the invalid request")
```
Comment (axthosarouris) [inline src/test/java/features/ScenarioCommonSteps.java:48]: moved from UserFt.java

File: src/main/java/no/unit/nva/handlers/AddUserHandlerForInternalService.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import no.unit.nva.database.DatabaseService;
+import no.unit.nva.model.UserDto;
+import nva.commons.exceptions.ApiGatewayException;
+import nva.commons.handlers.RequestInfo;
+import nva.commons.utils.Environment;
+import nva.commons.utils.JacocoGenerated;
+
+public class AddUserHandlerForInternalService extends AddUserHandler {
+
+    private final ApiKeyValidation apiKeyValidation;
```
Comment (janbj) [inline src/main/java/no/unit/nva/handlers/AddUserHandlerForInternalService.java:14]: ApiKeyValidator?

File: src/main/java/no/unit/nva/handlers/AddUserHandlerForInternalService.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    public AddUserHandlerForInternalService(Environment environment, DatabaseService databaseService,
+                                            AWSSecretsManager secretsManager) {
+        super(environment, databaseService);
+        this.apiKeyValidation = new ApiKeyValidation(secretsManager, environment);
+    }
+
+    @Override
+    protected UserDto processInput(UserDto input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        this.apiKeyValidation.authorizeAccess(requestInfo);
```
Comment (janbj) [inline src/main/java/no/unit/nva/handlers/AddUserHandlerForInternalService.java:30]: Shouldn't we look into how to do this in the API Gateway?

File: src/main/java/no/unit/nva/handlers/ApiKeyValidation.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+import no.unit.nva.exceptions.ForbiddenException;
+import nva.commons.handlers.RequestInfo;
+import nva.commons.utils.Environment;
+import nva.commons.utils.JacocoGenerated;
+import nva.commons.utils.JsonUtils;
+import nva.commons.utils.attempt.Failure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ApiKeyValidation {
```
Comment (janbj) [inline src/main/java/no/unit/nva/handlers/ApiKeyValidation.java:20]: Maybe this can be turned into a Lambda Authorizer?

**Pull request closed without merging**